### DISCUSSION TITLE
<小作业：planning>中关于约定输入的进一步确认

### DISCUSSION THREAD

**Discussion by DM0U** [physics-data/faq#52] [General] [answered]
小作业planning中有一些未声明的约定，但在样例中并未被违背。（实际上违背以下约定会造成一定的编程难度）
想确认以下约定是否会被违反：

1 任何表达式右侧只会产生一种反应产物
反例：
1 a + 1 b -> 1 c + 1 d
Q:
d 3

2 产物有唯一表达式，即所有产物只会在表达式右侧出现1次
反例：
1 a + 1 b -> 1 e
1 a + 2 c -> 1 e
1 c -> 1 b
Q:
e 3

2.1 产物没有唯一表达式，但能保证不存在循环表达式
反例：
1 a -> 1 b
1 b -> 1 c
1 c -> 1 a
1 d -> 1 b
Q:
a 3

2.2 产物没有唯一表达式，但合成有唯一解
反例：
1 a + 1 b -> 1 e
1 c + 1 d -> 1 e
Q:
e 3

2.3 产物没有唯一表达式，但不同答案间存在可以比较的最优解
例子（非反例）：
1 a + 1 b -> 1 e
2 a + 1 c -> 1 e
1 c -> 1 b
Q:
e 3

2.4 等式两侧不会出现同一物品
反例：
1 b -> 1 a
1 b + 1 a -> 3 a
Q:
a 3

**Comment by Fugoes** [comment, marked as answer]
对于 1，它并不符合题面（非形式化地）定义的文法，因为描述中右边不能出现 +；

对于 2，你提到的所有反例都不会出现；

总的来说，测试点除了第一个，都会使用 gen.py 生成的。
